File: 백준/Silver/1620. 나는야 포켓몬 마스터 이다솜/나는야 포켓몬 마스터 이다솜.py
N, M = map(int, input().split())
name_to_num = {}
num_to_name = {}

for i in range(1, N + 1):
    name = input()
    name_to_num[name] = i
    num_to_name[i] = name
    
for _ in range(M):
    query = input()
    if query.isdigit():
        print(num_to_name[int(query)])
    else:
        print(name_to_num[query])

File: 백준/Silver/1620. 나는야 포켓몬 마스터 이다솜/나는야 포켓몬 마스터 이다솜.js
const input = require('fs').readFileSync(0).toString().trim().split('\n');
const [N, M] = input[0].split(' ').map(Number);

const nameToNum = new Map();
const numToName = new Map();

for (let i = 1; i <= N; i++) {
    let name = input[i];
    nameToNum.set(name, i);
    numToName.set(i, name);
}

const result = [];
for (let i = N + 1; i <= N + M; i++) {
    let query = input[i];
    
    if (isNaN(Number(query))) {
        result.push(nameToNum.get(query));
    } else {
        result.push(numToName.get(Number(query)));
    }
}

console.log(result.join('\n'));